prestodb/dbapi.py

```diff
@@ -0,0 +1,232 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+
+This module implements the Python DBAPI 2.0 as described in
+https://www.python.org/dev/peps/pep-0249/ .
+
+Fetch methods returns rows as a list of lists on purpose to let the caller
+decide to convert then to a list of tuples.
+"""
+from future.standard_library import install_aliases
+install_aliases()
+import logging
+
+from typing import Any, List, Optional  # NOQA for mypy types
+
+from prestodb import constants
+import prestodb.exceptions
+import prestodb.client
+
+
+__all__ = ['connect', 'Connection', 'Cursor']
+
+
+apilevel = '2.0'
+threadsafety = 2
+
+logger = logging.getLogger(__name__)
+
+
+class Error(Exception):
+    pass
+
+
+class DatabaseError(Error):
```

Comment:
It is part of the DBAPI 2.0 specification but I just removed it as it is not used and there are other exceptions that are not defined. Error handling could be improved to better use these exceptions.